src/App.tsx

```diff
@@ -1,26 +1,179 @@
+import * as clientTodo from './api/todos';
+import { getFilteredTodos } from './utils/getFilteredTodos';
 
-const USER_ID = 0;
+import { ErrorMessage } from './types/ErrorMessage';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+
+import { TodoHeader } from './components/TodoHeader';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [activeStatus, setActiveStatus] = useState(Status.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage>(
+    ErrorMessage.Default,
+  );
+
+  const filteredTodos = useMemo(
+    () => getFilteredTodos(todos, activeStatus),
+    [todos, activeStatus],
+  );
+  const notCompletedTodos = useMemo(
+    () => todos.filter(todo => !todo.completed),
+    [todos],
+  );
+  const completedTodos = useMemo(
+    () => todos.filter(todo => todo.completed),
+    [todos],
+  );
+  const isToogleAll = completedTodos.length === todos.length;
+
+  const onAddTodo = (title: string) => {
+    setTempTodo({
+      id: 0,
+      title,
+      userId: clientTodo.USER_ID,
+      completed: false,
+    });
+
+    const newTodo: Omit<Todo, 'id'> = {
+      title,
+      userId: clientTodo.USER_ID,
+      completed: false,
+    };
+
+    return clientTodo
+      .createTodos(newTodo)
+      .then(todo => setTodos(currentTodos => [...currentTodos, todo]))
+      .catch(err => {
+        setErrorMessage(ErrorMessage.UnableToAdd);
+        throw err;
+      })
+      .finally(() => setTempTodo(null));
+  };
+
+  const onUpdateTodo = (todoToUpdate: Todo) => {
+    setLoadingTodos(prevTodos => [...prevTodos, todoToUpdate.id]);
+
+    return clientTodo
+      .updateTodo(todoToUpdate)
+      .then(updatedTodo => {
+        setTodos(currentTodos => {
+          const newTodos = [...currentTodos];
+          const index = newTodos.findIndex(todo => todo.id === updatedTodo.id);
+
+          newTodos.splice(index, 1, updatedTodo);
+
+          return newTodos;
+        });
+      })
+      .catch(err => {
+        setErrorMessage(ErrorMessage.UnableToUpdate);
+        throw err;
+      })
+      .finally(() => {
+        setLoadingTodos(prevTodos =>
+          prevTodos.filter(id => todoToUpdate.id !== id),
+        );
+      });
+  };
+
+  const onDeleteTodo = (todoId: number) => {
+    setLoadingTodos(prevTodos => [...prevTodos, todoId]);
+
+    clientTodo
+      .deleteTodo(todoId)
+      .then(() =>
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        ),
+      )
+      .catch(error => {
+        setErrorMessage(ErrorMessage.UnableToDelete);
+        throw error;
```

Comment:
You don't need to throw an error if you don't handle it on the next step. Fix in all places